src/main/java/fi/aalto/cs/intellij/activities/RequiredPluginsCheckerActivity.java

```diff
@@ -0,0 +1,213 @@
+ */
+public class RequiredPluginsCheckerActivity implements StartupActivity {
+
+  private static final Logger logger = LoggerFactory
+      .getLogger(RequiredPluginsCheckerActivity.class);
+
+  private final Map<String, String> requiredPluginNames = new HashMap<>();
+  private Map<String, String> missingOrDisabledPluginNames = new HashMap<>();
+  private List<IdeaPluginDescriptor> missingOrDisabledIdeaPluginDescriptors;
+  private List<IdeaPluginDescriptor> availableIdeaPluginDescriptors;
+
+  /**
+   * An actual startup work gets done here.
+   */
+  @Override
+  public void runActivity(@NotNull Project project) {
+    populateRequiredPluginNamesMap();
+    filterMissingOrDisabledPluginNames();
+    createListOfMissingOrDisabledPluginDescriptors();
+    checkPluginsStatusAndNotify();
+  }
+
+  /**
+   * Fills in the list of required plugin names.
+   *
+   * <p>Later, reading from the from configuration file might occur here.
+   */
+  private void populateRequiredPluginNamesMap() {
+    requiredPluginNames.put("Scala", "org.intellij.scala");
+  }
+
+  /**
+   * Filters out the plugin names that are missing from the current installation.
+   */
+  private void filterMissingOrDisabledPluginNames() {
+    missingOrDisabledPluginNames = requiredPluginNames
+        .entrySet()
+        .stream()
+        .filter(entry -> isPluginMissingOrDisabled(entry.getValue()))
+        .collect(toMap(Entry::getKey, Entry::getValue));
+  }
+
+  /**
+   * Predicate for checking the plugin status.
+   */
+  private static boolean isPluginMissingOrDisabled(String id) {
+    PluginId pluginId = PluginId.getId(id);
+    return !PluginManager.isPluginInstalled(pluginId) || PluginManager
+        .isDisabled(pluginId.getIdString());
+  }
+
+  /**
+   * If there are any plugins missing, creates a list of the plugin descriptors for them based on the
+   * publicly available ones.
+   */
+  private void createListOfMissingOrDisabledPluginDescriptors() {
+    if (missingOrDisabledPluginNames.size() > 0) {
+      getAvailablePluginsFromMainRepo();
+      missingOrDisabledIdeaPluginDescriptors = new ArrayList<>();
+      missingOrDisabledPluginNames.forEach((name, id) -> {
+        availableIdeaPluginDescriptors
+            .stream()
+            .filter(
+                availableDescriptor -> availableDescriptor.getPluginId().equals(PluginId.getId(id))
+            )
+            .findFirst()
+            .ifPresent(missingOrDisabledIdeaPluginDescriptors::add);
+      });
+    }
+  }
+
+  /**
+   * Get the list of all the available in the main JetBrains plugin repository.
+   *
+   * <p>Note! The descriptors for the not installed plugins do not exist in IJ until now.
+   */
+  private void getAvailablePluginsFromMainRepo() {
+    try {
+      availableIdeaPluginDescriptors = RepositoryHelper.loadPlugins(new BgProgressIndicator());
+    } catch (IOException ex) {
+      logger.error("Could not retrieve plugins data from main repository.", ex);
```

Comment:
We should do something else in addition to adding a log entry, e.g. a notification. There's bound to be people with e.g. internet down. The possibility of failing to download plugin information should be taken into account in runActivity-method.